### DISCUSSION TITLE
Using Retrivers

### DISCUSSION THREAD

**Discussion by sushantbhanot84** [patterns-ai-core/langchainrb#489] [Q&A]
I'm trying to implement a RAG system using ParentDocumentRetriever. I'm not able to find anything relevant in the code and readme about implementing a retriever.

**Comment by andreibondarev** [comment]
@sushantbhanot84 We're not necessarily aiming for parity with the Python Langchain but I'm open to ideas on how to integrate this in. Do you have any ideas on how we'd introduce this functionality and abstractions?

**Reply by sushantbhanot84** [reply]
You can have a separate module for retrievers containing classes for all the possible retrievers here: https://python.langchain.com/docs/modules/data_connection/retrievers/

**Reply by andreibondarev** [reply]
@sushantbhanot84 Hmm... yeah I'm open to create additional abstractions. Could you please describe your use-case in more detail?

I assume that you're uploading documents, chunking them and indexing the chunks into a vector search database. When you're retrieving them you'd like to run a similarity search -- match the individual chunk and retrieve the full parent documents themselves. Is this correct? It's also described [here](https://pub.towardsai.net/advanced-rag-techniques-an-illustrated-overview-04d193d8fec6) in the 2.4.2.
